src/components/FeatureStore/featureStore.util.js

```diff
@@ -576,34 +566,84 @@ export const fetchFeatureVectorRowData = async (
     }
   }))
 
-  const result = await fetchFeatureVector(
-    item.name,
-    match.params.projectName
-  ).catch(error => {
+  const result = await fetchFeatureVector(item.name, item.project).catch(
+    error => {
+      setPageData(state => ({
+        ...state,
+        selectedRowData: {
+          ...state.selectedRowData,
+          [item.name]: {
+            ...state.selectedRowData[item.name],
+            error,
+            loading: false
+          }
+        }
+      }))
+    }
+  )
+
+  if (result?.length > 0) {
+    setYamlContent(state => ({ ...state, selectedRowData: result }))
     setPageData(state => ({
       ...state,
       selectedRowData: {
-        ...state.selectedRowData,
         [item.name]: {
-          ...state.selectedRowData[item.name],
-          error,
+          content: [...parseFeatureVectors(result)],
+          error: null,
           loading: false
         }
       }
     }))
-  })
+  }
+}
 
-  if (result?.length > 0) {
-    setYamlContent(state => ({ ...state, selectedRowData: result }))
+export const fetchDataSetRowData = async (
+  fetchDataSet,
+  item,
+  setPageData,
+  setYamlContent
+) => {
+  setPageData(state => ({
+    ...state,
+    selectedRowData: {
+      ...state.selectedRowData,
+      [item.db_key]: {
+        loading: true
+      }
+    }
+  }))
+
+  const result = await fetchDataSet(item).catch(error => {
```

Comment:
Combining `async`/`await` with `Promise.prototype.then`/`Promise.prototype.catch`/`Promise.prototype.finally` seems confusing to me.
Why not stick to _either_ one approach _or_ the other?
`async`/`await` are usually used with `try...catch` statements like this:

```js
try {
  const result = await doSomeOperation(args)
  // do something with `result`
} catch (error) {
  // do something with `error`
} finally { // optional
  // do some cleanup if necessary
}
```
The equivalent with direct `Promise` API:
```js
doSomeOperation(args)
  .then(result => {
    // do something with `result`
  })
  .catch(error => {
    // do something with `error`
  })
  .finally(() => { // optional
    // do some cleanup if necessary
  })
```